lazy_theta_star_b/src/lazy_theta_star_b.cpp

```diff
@@ -0,0 +1,529 @@
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Anshumaan Singh
+ *********************************************************************/
+#include "lazy_theta_star_b/lazy_theta_star_b.h"
+#include "nav2_util/node_utils.hpp"
+#include "chrono"
+
+namespace lazy_theta_star_b {
+    void LazyThetaStarB::configure(rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+                                   std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+                                   std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros) {
+        node_ = parent;
+        name_ = name;
+        tf_ = tf;
+        costmap_ = costmap_ros->getCostmap();
+        global_frame_ = costmap_ros->getGlobalFrameID();
+
+        nav2_util::declare_parameter_if_not_declared(
+                node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+                        8));
+
+        node_->get_parameter(name_ + ".how_many_corners", how_many_corners_);
+
+        if (how_many_corners_ != 8)
+            how_many_corners_ = 4;
+
+        nav2_util::declare_parameter_if_not_declared(
+                node_, name_ + ".interpolation_dist", rclcpp::ParameterValue(
+                        0.1));
+
+        node_->get_parameter(name_ + ".interpolation_dist", interpolation_dist);
+
+    }
+
+    void LazyThetaStarB::cleanup() {
+        RCLCPP_INFO(
+                node_->get_logger(), "CleaningUp plugin %s of type LazyThetaStarB",
+                name_.c_str());
+    }
+
+    void LazyThetaStarB::activate() {
+        RCLCPP_INFO(
+                node_->get_logger(), "Activating plugin %s of type LazyThetaStarB",
+                name_.c_str());
+    }
+
+    void LazyThetaStarB::deactivate() {
```

Comment:
We don't have to go over all the linting issues in this round, but just be aware, you have a bunch of linting issues that will need to be resolved (number of spaces, spaces between the `&` and `*` etc.

Reply:
You can run the linters locally with `ament_cpplint` and `ament_uncrustify`, for uncrustify, `--reformat` will reformat it automatically to be in compliance, cpplint needs manual attention